numpy/linalg/linalg.py

```diff
@@ -918,14 +918,14 @@ def qr(a, mode='reduced'):
     # determine optimal lwork
     lwork = 1
     work = zeros((lwork,), t)
-    results = lapack_routine(m, mc, mn, q, m, tau, work, -1, 0)
+    results = lapack_routine(m, mc, mn, q, max(1, m), tau, work, -1, 0)
     if results['info'] != 0:
         raise LinAlgError('%s returns %d' % (routine_name, results['info']))
 
     # compute q
-    lwork = int(abs(work[0]))
+    lwork = max(1, int(abs(work[0])))
```

Comment:
I'd be surprised if this is necessary for the work spaces, since lapack should be populating them with whatever it really needs

Reply:
It's not. But my problem was having to pass in `lwork=0` which was invalid (it had to be at least... oh shit I may have made a mistake...)

http://www.netlib.org/lapack/explore-3.1.1-html/dgeqrf.f.html
http://www.netlib.org/lapack/explore-3.1.1-html/zungqr.f.html

Tests pass but I'm going to go fix it...